app/src/main/java/com/example/kotlinpps/mvp/view/MainActivityView.kt

```diff
@@ -1,16 +1,54 @@
 package com.example.kotlinpps.mvp.view
 
 import android.app.Activity
+import android.content.Intent
 import android.widget.Toast
+import com.example.kotlinpps.GOOGLE_OPTIONS
+import com.example.kotlinpps.GOOGLE_SIGN_IN
 import com.example.kotlinpps.R
+import com.example.kotlinpps.activity.MainScreenActivity
+import com.example.kotlinpps.activity.SignInActivity
 import com.example.kotlinpps.mvp.contract.MainActivityContract
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
 import kotlinx.android.synthetic.main.activity_main.*
 import java.lang.ref.WeakReference
 
 class MainActivityView(activity: Activity) : MainActivityContract.View {
     private val activity: WeakReference<Activity> = WeakReference(activity)
 
-    override fun onButtonPressed() {
-        Toast.makeText(activity.get(), R.string.app_name,Toast.LENGTH_LONG).show()
+    override fun onGoogleButtonPressed(googleSignInOptions: GoogleSignInOptions) {
+        val signInIntent = GoogleSignIn.getClient(activity.get()!!, googleSignInOptions).signInIntent
+        activity.get()!!.startActivityForResult(signInIntent, GOOGLE_SIGN_IN)
+    }
+
+    override fun onExternalFailure() {
+        Toast.makeText(activity.get(), R.string.external_log_in_error, Toast.LENGTH_SHORT).show()
+    }
+
+    override fun onLocalEmailFailure() {
+        Toast.makeText(activity.get(), R.string.local_email_error, Toast.LENGTH_SHORT).show()
+    }
+
+    override fun onLocalPasswordFailure() {
```

Comment:
you could use a single method to show a toast with the message as a parameter.

Reply:
Done